### PR TITLE
fix: criteria search so match uris, not labels

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-515
WIP

- [x] URI should be used instead of labels for the search terms
- [x]  Frontend code adapted

Related PR: 

- [x] https://github.com/oat-sa/lib-tao-elasticsearch/pull/47
- [x] https://github.com/oat-sa/tao-core-ui-fe/pull/331

Helping PR https://github.com/oat-sa/tao-core/pull/3028

### CODE REVIEW

File: models/classes/search/index/DocumentBuilder/IndexDocumentBuilder.php [hunk 1]
```diff
@@ -235,18 +236,19 @@ protected function getDynamicProperties(array $classes, Resource $resource): Ite
...
                 $propertyTypeArray = explode('#', $propertyTypeUri, 2);
                 $propertyTypeId = end($propertyTypeArray);
-                $customPropertyLabel = $property->getLabel();
 
                 if (false === $propertyTypeId) {
                     continue;
                 }
 
-                $fieldName = $propertyTypeId . '_' . \tao_helpers_Slug::create($customPropertyLabel);
+                $fieldName = $propertyTypeId . '_' . tao_helpers_Uri::encode($property->getUri());
```
Comment (gabrielfs7) [inline models/classes/search/index/DocumentBuilder/IndexDocumentBuilder.php:244]: Do we need to change `GenerisIndexDocumentBuilderTest` for this?

File: actions/class.ClassMetadata.php [hunk 2]
```diff
@@ -36,9 +38,14 @@ public function get(
         ClassMetadataSearcherProxy $classMetadataSearcher
     ): void {
         $this->setSuccessJsonResponse(
-            $classMetadataSearcher->findAll(
-                $classMetadataSearchRequestHandler->handle($request)
-            )
+
+//            new ClassInformation(
```
Comment (gabrielfs7) [inline actions/class.ClassMetadata.php:42]: will it be removed?

Comment (boajer) [reply actions/class.ClassMetadata.php:42]: Will be adapted soon, as FE changes just arrived

Comment (gabrielfs7) [reply actions/class.ClassMetadata.php:42]: Can we avoid a breaking change and add required notes in the existing object?

File: models/classes/Lists/Business/Domain/ClassInformation.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public function __construct(ClassCollection $classCollection, array $searchMapping)
+    {
+        $this->classDefinition = $classCollection;
+        $this->criteriaMapping = $searchMapping;
+    }
+
+    public function jsonSerialize(): array
+    {
+        return [
+            'classDefinition' => $this->classDefinition,
```
Comment (gabrielfs7) [inline models/classes/Lists/Business/Domain/ClassInformation.php:44]: Do you think `classDefinition` is a good name to represent a collection os class metadata?

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]
LGTM

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/Lists/Business/Domain/ClassInformation.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+declare(strict_types=1);
+
+namespace oat\tao\model\Lists\Business\Domain;
+
+use JsonSerializable;
+
+class ClassInformation implements JsonSerializable
+{
+
+    private $classDefinition;
```
Comment (gabrielfs7) [inline models/classes/Lists/Business/Domain/ClassInformation.php:31]: MIssing type: `ClassCollection`

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**